### PR TITLE
Adding Link to ADS Extension Generator Documentation in Extensions Menu

### PR DESCRIPTION
**Pull request by anjalia**
Adds a link to the [ADS Extension Generator Documentation](https://docs.microsoft.com/sql/azure-data-studio/extension-authoring) from the Extensions Menu 

extensionMenu

### CODE REVIEW

File: src/vs/workbench/contrib/extensions/browser/extensionsActions.ts [hunk 1]
```diff
@@ -3494,6 +3494,25 @@ export class InstallLocalExtensionsInRemoteAction extends Action {
 	}
 }
 
+export class OpenExtensionAuthoringDocsAction extends Action {
```
Comment (Charles-Gagnon) [inline src/vs/workbench/contrib/extensions/browser/extensionsActions.ts:3497]: Put this in https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/extensions/browser/extensionsActions.ts#L1

File: src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ import {
 	ClearExtensionsInputAction, ChangeSortAction, UpdateAllAction, CheckForUpdatesAction, DisableAllAction, EnableAllAction,
 	EnableAutoUpdateAction, DisableAutoUpdateAction, ShowBuiltInExtensionsAction, InstallVSIXAction, SearchCategoryAction,
 	/*RecentlyPublishedExtensionsAction, */ShowInstalledExtensionsAction, ShowOutdatedExtensionsAction, ShowDisabledExtensionsAction,
-	ShowEnabledExtensionsAction, PredefinedExtensionFilterAction
+	ShowEnabledExtensionsAction, PredefinedExtensionFilterAction, OpenExtensionAuthoringDocsAction
```
Comment (Charles-Gagnon) [inline src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts:25]: Add `// {{ SQL CARBON EDIT }}` comment on this line

File: src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts [hunk 3]
```diff
@@ -555,6 +555,7 @@ export class ExtensionsViewPaneContainer extends ViewPaneContainer implements IE
 
 		actions.push(new Separator());
 		actions.push(this.instantiationService.createInstance(InstallVSIXAction, InstallVSIXAction.ID, InstallVSIXAction.LABEL));
+		actions.push(this.instantiationService.createInstance(OpenExtensionAuthoringDocsAction, OpenExtensionAuthoringDocsAction.ID, OpenExtensionAuthoringDocsAction.LABEL));
```
Comment (Charles-Gagnon) [inline src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts:558]: Add `// {{ SQL CARBON EDIT }}` comment on this line

File: src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts [hunk 4]
```diff
@@ -160,6 +162,8 @@ actionRegistry.registerWorkbenchAction(SyncActionDescriptor.from(DisableAutoUpda
 actionRegistry.registerWorkbenchAction(SyncActionDescriptor.from(InstallSpecificVersionOfExtensionAction), 'Install Specific Version of Extension...', ExtensionsLabel);
 actionRegistry.registerWorkbenchAction(SyncActionDescriptor.from(ReinstallAction), 'Reinstall Extension...', localize({ key: 'developer', comment: ['A developer on Code itself or someone diagnosing issues in Code'] }, "Developer"));
 
+actionRegistry.registerWorkbenchAction(SyncActionDescriptor.from(OpenExtensionAuthoringDocsAction), 'Extensions: Author an Extension...', ExtensionsLabel);
```
Comment (Charles-Gagnon) [inline src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts:165]: Put this in https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts#L1

File: src/vs/workbench/contrib/extensions/browser/extensionsActions.ts [hunk 5]
```diff
@@ -3494,6 +3494,25 @@ export class InstallLocalExtensionsInRemoteAction extends Action {
...
+
+	static readonly ID = 'workbench.extensions.action.openExtensionAuthoringDocs';
+	static readonly LABEL = localize('openExtensionAuthoringDocs', "Author an Extension...");
```
Comment (Charles-Gagnon) [inline src/vs/workbench/contrib/extensions/browser/extensionsActions.ts:3500]: I think `Create an Extension` sounds better, but that's just a suggestion

**Comment by anjalia** [comment]
Sure @Charles-Gagnon , description updated with a screenshot

**Comment by anjalia** [comment]
Thank you for the review -- we hadn't thought of adding it to the Welcome page, but I'll reach out to Rayden about it

**Pull request merged**